### DISCUSSION TITLE
Offensive Language in Suggested Code

### DISCUSSION THREAD

**Discussion by ghost** [community/community#14226] [Copilot Conversations]
Blurred for obvious reasons.
![Code_43dB4BM54E](https://user-images.githubusercontent.com/16302135/161484400-b8c32e86-a52d-422a-85e2-f381986bf6f4.png)

I was attempting to make a reference to a funny (but quite inappropriate) YouTube video since the code I'm writing is just throw-away code, and this racial slur was suggested to me.

**Comment by Urethramancer** [comment]
I've found that it learns my preferred swearing over time, so it might sort itself out over time :P
(But yeah, perhaps they should have at least SOME profanity filtering there!)

**Comment by karafra** [comment]
Dont swear at it and it will fix itself over time 🤷🏼

**Comment by derekxmartin** [comment]
Just a thought: is it possible that CoPilot is not compliant with the OpenAI API’s requirement for content moderation?

> [Standard safety requirements (when applications have difficult-to-mitigate misuse)](https://beta.openai.com/docs/usage-guidelines/standard-safety-requirements-when-applications-have-difficult-to-mitigate-misuse)
In general, applications with difficult-to-mitigate misuse are required to adhere to the conditions below.

> Standard requirements:

* Implement the free [OpenAI Content Filter](https://beta.openai.com/docs/engines/content-filter) to prevent Unsafe (CF=2) completions from being returned to the end-user.

I think this would go along way in terms of addressing user concerns. If this is already implemented, that’s great. And in that case there probably just needs to be more education to the public about GPT-3’s inherit biases and risks so that the proper blame can be attributed.
